libsatadrivers/src/ahci.c

```diff
@@ -0,0 +1,956 @@
+/*
+ * Copyright 2019, DornerWorks
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * A lot of this code was based on code found at http://wiki.osdev.org/AHCI
+ * which is in the public domain.
+ *
+ * @TAG(DORNERWORKS_BSD)
+ *
+ */
+
+#include <satadrivers/ide.h>
+#include <satadrivers/ahci.h>
+#include <unistd.h>
+#include <string.h>
+#include <sel4/sel4.h>
+#include <platsupport/delay.h>
+
+#include <satadrivers/ahci_types.h>
+#include <utils/zf_log.h>
+
+#define GET_LOWER_32BITS(val) (uint32_t)((uint64_t)(val) & 0xFFFFFFFFLL)
```

Comment:
why not simply `((uint32_t)(val))`

Reply:
Added assurance I suppose. Not too sure, I didn't write it.